traitsui/qt4/html_editor.py

```diff
@@ -26,6 +26,29 @@
 
 from .editor import Editor
 
+
+# Subclass of QWebPage for QtWebEngine support
+
+class ExternallyOpeningWebPage(QtWebKit.QWebPage):
+    """ QWebEnginePage subclass that opens links in system browser
+
+    This subclass is only used when we are given a QWebEnginePage which is
+    pretending to be a QWebPage and we want the open_external feature
+    of the Editor.
+
+    This overrides the acceptNavigationRequest method to open links
+    in an external browser.  All other navigation requests are handled
+    internally as per the base class.
+    """
+
+    def acceptNavigationRequest(self, url, type, isMainFrame):
+        if type == QtWebKit.QWebPage.NavigationTypeLinkClicked:
+            webbrowser.open_new(url.toString())
+            return False
```

Comment:
I believe this `False` is because of this from the documentation on `acceptNavigationRequest`: "Note: The loading process is started and the loadStarted() signal is emitted before the request is accepted or rejected. Therefore, a loadFinished() signal that returns false is to be expected even after delegating the request."?  Is that correct?  This comment from the docs and this `return False` were the only things I still am not 100% clear on

Reply:
Not quite - it's part of the API of acceptNavigationRequest.

>  If the function returns true, the navigation request is accepted and url is loaded.

By returning `False` we are signalling that we are handling this case and that the page should do nothing more.